app/api/entities/v2_support.ts

```diff
@@ -0,0 +1,124 @@
+import { ObjectId } from 'mongodb';
+
+import templates from 'api/templates';
+import { DefaultEntitiesDataSource } from 'api/entities.v2/database/data_source_defaults';
+import { DenormalizationService } from 'api/relationships.v2/services/service_factories';
+import { search } from 'api/search';
+import { DefaultSettingsDataSource } from 'api/settings.v2/database/data_source_defaults';
+import { MongoTransactionManager } from 'api/common.v2/database/MongoTransactionManager';
+import { getClient } from 'api/common.v2/database/getConnectionForCurrentTenant';
+import { objectIndex } from 'shared/data_utils/objectIndex';
+import { EntitySchema } from 'shared/types/entityType';
+import { MetadataSchema } from 'shared/types/commonTypes';
+import { propertyTypes } from 'shared/propertyTypes';
+import { TemplateSchema } from 'shared/types/templateType';
+import { DefaultRelationshipDataSource } from 'api/relationships.v2/database/data_source_defaults';
+
+const entityTypeCheck = (
+  entity: EntitySchema
+): entity is EntitySchema & {
+  template: ObjectId;
+  sharedId: string;
+  metadata: MetadataSchema;
+} => !!entity.template && !!entity.sharedId && !!entity.metadata;
+
+const performNewRelationshipQueries = async (entities: EntitySchema[]) => {
+  const transactionManager = new MongoTransactionManager(getClient());
+  if (!(await DefaultSettingsDataSource(transactionManager).readNewRelationshipsAllowed())) {
+    return;
+  }
+
+  const templateIdToProperties = objectIndex(
+    await templates.get({ _id: entities.map(e => e.template) }),
+    t => t._id.toString(),
+    t => t.properties.filter((prop: any) => prop.type === 'newRelationship')
+  );
```

Comment:
templateIdToProperties is not properly typed as { [k: string]: any }, is there any way to infer this type properly ? worst case we can just  type the variable inline to avoid the "any" types from this point on.

Reply:
Makes sense. I will try to type the function properly. That should be rather quick. The possible problem I foresee is the rest of the usages already in the code. Hopefully they don't break :D